Chore: Fix errors on `npm install` on mac with M1 chip

samanthaandrews opened the pull request:
I was unable to run `npm install` on a fresh clone of this project (i got a lot of node-gyp errors from the `canvas` package?). This PR ensures new users to the project can successfully run `npm install` and all of the other npm commands that the project requires.

### This PR does the following:
- Removes the package-lock.json and performs a fresh install to fix dependency errors
- Updates React, Chakra, Reservoir Design System, Playwright, and Cucumber to the latest versions
- Removes several unused packages
- Turns cucumber.js into cucumber.json because for some reason `worldParameters` were no longer being picked up in the .js file
- Updates snapshot tests
- Prettier fixes
- Cleans up GH Actions CI setup node steps

### Testing requirements & instructions: 
-  I ran `npm run cucumber` and all of the end-to-end tests passed, so I feel confident about the changes in this PR

samanthaandrews commented on cucumber.json:
```diff
@@ -0,0 +1,10 @@
+    "paths": ["playwright/features/**/*.feature"],
+    "import": ["playwright/tests/**/*.ts", "playwright/support/**/*.ts"],
+    "worldParameters": {
+      "appUrl": "https://drb-qa.nypl.org/",
```
I changed the `appUrl` to the QA one, since these tests should run on QA so as not to overload our production servers cc @chrismulholland

jackiequach commented on .github/workflows/ci.yml:
```diff
@@ -51,7 +51,7 @@ jobs:
 
       - name: Install Dependencies
-        run: npm run install:ci
+        run: npm ci
```
this is cool, didn't know about this command.

samanthaandrews commented:
@jackiequach can you pull this branch down locally, delete `node_modules`, and try to run `npm install` and make sure the project builds and tests pass for you? I don't want to merge this unless it works for you. (I have a mac with an M1 chip and I know that sometimes this can cause different install issues)

jackiequach commented:
@samanthaandrews installs, builds, and runs fine. The `About` and `License` snapshot and Advanced Search clear › clears all searches tests are failing.

samanthaandrews commented:
> @samanthaandrews installs, builds, and runs fine. The `About` and `License` snapshot and Advanced Search clear › clears all searches tests are failing.

Thank you for testing this. I do think the AdvancedSearch test failing is an actual regression (text inputs aren't clearing after clicking the `clear` button). I'm going to move this to draft state until I get that issue sorted out!

samanthaandrews commented:
FYI @jackiequach I fixed those test regressions by installing the latest version of the design system so I'll _finally_ merge this PR today! Once I do, you will probably want to do run `npm i` since I updated some package version numbers.
